src/device.cpp

```diff
@@ -48,6 +48,13 @@ void cvk_device::init_vulkan_properties(VkInstance instance) {
     m_subgroup_properties.sType =
         VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_SUBGROUP_PROPERTIES;
 
+    //--- Get maxMemoryAllocationSize for figuring out the  max single buffer
+    // allocation size.
+    m_maintenance3_properties.sType =
+        VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_MAINTENANCE_3_PROPERTIES;
+    m_maintenance3_properties.pNext = nullptr;
```

Comment:
```suggestion
```

You don't need this, pNext pointers are all handled automatically by the code below :).

Reply:
done